python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,258 @@
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration
+        """
+
+        # Getting actuators to be tested.
+        self.actuators_to_test = (
+            self.m1m3_actuator_ids if config.actuators == "all" else config.actuators
+        )
+
+    def set_metadata(self, metadata):
+        """Set metadata."""
+
+        # Getting total number of secondary actuators to be tested
+        total_tested_secondary = sum(
+            [
+                1
+                for actuator in self.actuators_to_test
+                if self.has_secondary_actuator(actuator)
+            ]
+        )
+
+        # Setting metadata
+        metadata.duration = self.time_one_bump * (
+            len(self.actuators_to_test) + total_tested_secondary
+        )
+
+    def has_secondary_actuator(self, actuator_id: int) -> bool:
+        """Determines whether a given actuator has a
+        secondary axis or not.
+        """
+
+        return actuator_id in self.m1m3_secondary_actuator_ids
+
+    async def run(self):
+        await self.assert_feasibility()
+        start_time = time.time()
+
+        # Get M1M3 detailed state
+        detailed_state = DetailedState(
+            (
+                await self.mtcs.rem.mtm1m3.evt_detailedState.aget(
+                    timeout=self.mtcs.fast_timeout,
+                )
+            ).detailedState
+        )
+        self.log.info(f"Current M1M3 detailed state: {detailed_state!r}.")
+
+        # Put M1M3 in engineering mode
+        await self.mtcs.enter_m1m3_engineering_mode()
+
+        failed_actuators = []
+        for i, actuator_id in enumerate(self.actuators_to_test):
+            await self.mtcs.assert_all_enabled()
+            await self.checkpoint(f"Running bump test on FA ID: {actuator_id}.")
+
+            # Determine if actuator has a secondary actuator
+            secondary_exist = self.has_secondary_actuator(actuator_id)
+
+            try:
+                await self.mtcs.run_m1m3_actuator_bump_test(
+                    actuator_id=actuator_id,
+                    primary=True,
+                    secondary=secondary_exist,
+                )
+            # raise a RuntimeError if the bump test fails
+            # and continue loop to next actuator_id
+            except RuntimeError:
+                failed_actuators.append(actuator_id)
+                self.log.exception(f"Failed to run bump test on FA ID {actuator_id}.")
+                continue
+
+            # Getting test status
+            (
+                fa_id,
+                primary_status,
+                secondary_status,
+            ) = await self.mtcs.get_m1m3_bump_test_status()
+
+            if not secondary_exist:
+                self.log.info(
+                    f"Bump test done for {i + 1} of {len(self.actuators_to_test)}"
+                    f"Bump test on FA ID {fa_id}. Primary FA: "
+                    f"{primary_status.upper()}."
```

Comment:
If the primary_status is FAILED, you shall add that FA to failed FAs. This might negate need for running separate bump tests mentioned above.